x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -141,8 +114,55 @@ export class MapsAppView extends React.Component {
         filterManager.removeFilter(filter);
       }
     });
+
+    getCoreChrome().setBreadcrumbs([]);
+  }
+
+  _hasUnsavedChanges() {
+    return this.props.hasUnsavedChanges(this.props.savedMap, this.state.initialLayerListConfig);
   }
 
+  _setBreadcrumbs = () => {
+    getCoreChrome().setBreadcrumbs([
+      {
+        text: i18n.translate('xpack.maps.mapController.mapsBreadcrumbLabel', {
+          defaultMessage: 'Maps',
+        }),
+        onClick: () => {
+          if (this._hasUnsavedChanges()) {
+            const navigateAway = window.confirm(unsavedChangesWarning);
+            if (navigateAway) {
+              goToSpecifiedPath('/');
+            }
+          } else {
+            goToSpecifiedPath('/');
+          }
+        },
+      },
+      { text: this.props.savedMap.title },
+    ]);
+  };
```

Comment:
I'm not sure if we should attempt this in this PR or a future one maybe, but I wonder if we should keep `_hasUnsavedChanges` and `_setBreadcrumbs` in a separate file initialized with a `savedMap` by `load_map_and_render`. That would allow `load_map_and_render` be more of a "prep phase" and descendant components could call ` _setBreadcrumbs` with new `text` args. The most limiting piece to doing this would be `this.state.initialLayerListConfig` which itself ultimately relies on `_getInitialLayersFromUrlParam` which also is pretty standalone and could be moved.

Reply:
We could discuss such changes for the future but don't think this PR should include such changes.